src/main/java/au/gov/nla/pickslip/service/FolioService.java

```diff
@@ -248,4 +306,118 @@ protected JsonNode folioApiGetRequests() throws IOException {
   }
+
+  public FolioRequest folioApiGetRequestById(final String requestId) throws IOException {
+    JsonNode folioRequestJson =
+        new FOLIORequestsRetrieverAPI(folioOkapiCredentials).getRequestById(requestId);
+
+    JsonNode requestDateNode = folioRequestJson.at("/requestDate");
+    ZonedDateTime requestDate =
+        requestDateNode.isNull()
+            ? null
+            : ZonedDateTime.parse(
+            requestDateNode.asText(), DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+
+    assert requestDate != null;
+    ZonedDateTime localRequestDate =
+        requestDate.withZoneSameInstant(ZoneId.systemDefault());
+
+    return
+        new FolioRequest(
+            folioRequestJson.at("/id")
+                .asText(null),
+            localRequestDate,
+            folioRequestJson.at("/patronComments")
+                .asText(null),
+            folioRequestJson.at("/itemId")
+                .asText(null),
+            folioRequestJson.at("/instanceId")
+                .asText(null),
+            folioRequestJson.at("/requesterId")
+                .asText(null),
+            folioRequestJson.at("/status")
+                .asText(null),
+            folioRequestJson.at("/cancellationAdditionalInformation")
+                .asText(null),
+            folioRequestJson.at("/position")
+                .asText(null),
+            new FolioRequest.Instance(folioRequestJson.at("/instance/title")
+                .asText(null)),
+            new FolioRequest.Item(
+                folioRequestJson.at("/item/barcode")
+                    .asText(null),
+                folioRequestJson.at("/item/callNumber")
+                    .asText(null),
+                new FolioRequest.Item.Location(
+                    folioRequestJson.at("/item/location/name")
+                        .asText(null),
+                    folioRequestJson.at("/item/location/code")
+                        .asText(null))),
+            new FolioRequest.Requester(
+                folioRequestJson.at("/requester/barcode")
+                    .asText(null),
+                folioRequestJson.at("/requester/patronGroupGroup")
+                    .asText(null)),
+            toList(folioRequestJson.at("/tagList")
+                .elements()));
+  }
+
+  public void updateRequest(final RequestNoteDto requestNoteDto) throws IOException {
+    FOLIORequestsRetrieverAPI folioRequestsRetrieverAPI =
+        new FOLIORequestsRetrieverAPI(folioOkapiCredentials);
+    JsonNode folioRequestJson =
+        folioRequestsRetrieverAPI.getRequestById(requestNoteDto.getRequestId());
+    if (folioRequestJson != null) {
+      ((ObjectNode) folioRequestJson).put("cancellationAdditionalInformation",
+          requestNoteDto.getCancellationAdditionalInformation());
+      folioRequestsRetrieverAPI.updateRequest(requestNoteDto.getRequestId(), folioRequestJson);
+    }
+    else {
+      log.error("Could not retrieve request with id: {}", requestNoteDto.getRequestId());
```

Comment:
Should this result in feedback to the user?  (Imagining it'll fail silently apart from log file message.)